sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/ByteBuddyDoFnInvokerFactory.java

```diff
@@ -213,11 +244,18 @@ TrackerT invokeNewTracker(RestrictionT restriction) {
         "Signature is for class %s, but fn is of class %s",
         signature.fnClass(),
         fn.getClass());
+
     try {
       @SuppressWarnings("unchecked")
-      DoFnInvoker<InputT, OutputT> invoker =
-          (DoFnInvoker<InputT, OutputT>)
+      DoFnInvokerBase<InputT, OutputT, DoFn<InputT, OutputT>> invoker =
+          (DoFnInvokerBase<InputT, OutputT, DoFn<InputT, OutputT>>)
               getByteBuddyInvokerConstructor(signature).newInstance(fn);
+
+      for (OnTimerMethod onTimerMethod : signature.onTimerMethods().values()) {
```

Comment:
Ideally, we'd generate the byte-code to do the dispatching, so that we could build it once (when we make the invoker class) rather than repopulating for every `fn`. But, this shouldn't have too high of an overhead (especially with `fn` reuse). Since we should be able to make improvements here if necessary (should be backwards compatible) I don't think we need to do this now.